src/chromap.cc

```diff
@@ -426,12 +435,20 @@ bool Chromap::CorrectBarcodeAt(uint32_t barcode_index,
     const char *barcode_qual = barcode_batch.GetSequenceQualAt(barcode_index);
     std::vector<BarcodeWithQual> corrected_barcodes_with_quals;
     uint64_t mask = (uint64_t)3;
-    for (uint32_t i = 0; i < barcode_length; ++i) {
+    uint32_t i_start = 0;
+    uint32_t i_end = barcode_length;
+    uint32_t ti_limit = 3;
+    if (N_pos.size() > 0) {
+      i_start = N_pos[0];
+      i_end = N_pos[0] + 1;
+      ti_limit = 4;
```

Comment:
And the reason we change it to 4 is that we also want to see whether the original A is a good correction?

Reply:
Right.